pwiz/analysis/spectrum_processing/SpectrumList_IonMobility.cpp

```diff
@@ -170,15 +182,31 @@ PWIZ_API_DECL double SpectrumList_IonMobility::ccsToIonMobility(double ccs, doub
     }
 }
 
-PWIZ_API_DECL std::pair<int, int> SpectrumList_IonMobility::sonarMzToDriftBinRange(int function, float precursorMz, float precursorTolerance) const
+///
+/// Deal with Waters SONAR equipment, which filters an m/z range using its ion mobility hardware and reports the data as if it were ion mobility
+///
+PWIZ_API_DECL bool SpectrumList_IonMobility::isWatersSonarData() const
 {
-    if (equipment_ != IonMobilityEquipment::WatersSonar)
-        throw runtime_error("SpectrumList_IonMobility::sonarMzToDriftBinRange] function only works on Waters SONAR data");
+    return equipment_ == IonMobilityEquipment::WatersSonar;

```

Comment:
Definitely need the extra else branch above to set the equipment_ parameter for the new SONAR scheme, or else SONAR mzMLs certainly won't work with this function.

Reply:
Agreed. As I say, not yet worrying about that for mzML files.